optihood/energy_network.py

```diff
@@ -216,29 +213,37 @@ def get_nodal_data(func, data: _tp.Union[pd.ExcelFile, pd.DataFrame, dict]):
             func = {get_data_from_df}            
         """
         nodes_data = {
-            NodeKeys.buses.value: func(data, NodeKeys.buses.value),
-            NodeKeys.grid_connection.value: func(data, NodeKeys.grid_connection.value),
-            NodeKeys.commodity_sources.value: func(data, NodeKeys.commodity_sources.value),
-            NodeKeys.solar.value: func(data, NodeKeys.solar.value),
-            NodeKeys.transformers.value: func(data, NodeKeys.transformers.value),
-            NodeKeys.demand.value: func(data, NodeKeys.demand.value),
-            NodeKeys.storages.value: func(data, NodeKeys.storages.value),
-            NodeKeys.stratified_storage.value: func(data, NodeKeys.stratified_storage.value),
-            NodeKeys.profiles.value: func(data, NodeKeys.profiles.value)
+            _ent.NodeKeys.buses.value: func(data, _ent.NodeKeys.buses.value),
+            _ent.NodeKeys.grid_connection.value: func(data, _ent.NodeKeys.grid_connection.value),
+            _ent.NodeKeys.commodity_sources.value: func(data, _ent.NodeKeys.commodity_sources.value),
+            _ent.NodeKeys.solar.value: func(data, _ent.NodeKeys.solar.value),
+            _ent.NodeKeys.transformers.value: func(data, _ent.NodeKeys.transformers.value),
+            _ent.NodeKeys.demand.value: func(data, _ent.NodeKeys.demand.value),
+            _ent.NodeKeys.storages.value: func(data, _ent.NodeKeys.storages.value),
+            _ent.NodeKeys.stratified_storage.value: func(data, _ent.NodeKeys.stratified_storage.value),
+            _ent.NodeKeys.profiles.value: func(data, _ent.NodeKeys.profiles.value)
         }
         return nodes_data
 
+    @staticmethod
+    def get_values_from_dataframe(df: pd.DataFrame, identifier: str, identifier_column: str, desired_column: str):
```

Comment:
This is a very common implementation in this part.
Putting it into a function should make this part more readable.